terminusdb_client/woqlclient/woqlClient.py

```diff
@@ -805,13 +842,14 @@ def query(self, woql_query, commit_msg=None, file_dict=None):
             query_obj = self._generate_commit(commit_msg)
         else:
             query_obj = {}
-        if type(woql_query) != dict and hasattr(woql_query, "to_dict"):
-            woql_query = woql_query.to_dict()
-        woql_query["@context"] = self.conCapabilities.get_context_for_outbound_query(
+        if isinstance(woql_query, WOQLQuery):
+            request_woql_query = woql_query.to_dict()
+        request_woql_query["@context"] = self.conCapabilities.get_context_for_outbound_query(
```

Comment:
In a few places like here (and line 795 for instance), the function argument is modified in such a way that type checking breaks. I've updated the code with a local variable instead of changing the function argument, which I think is better practice.

Reply:
I would say yes if WOQLQuery can be imported.